src/Resources/assets/controllers/accordion_controller.js

```diff
@@ -1,70 +1,67 @@
-
-            nextSiblings.forEach((sibling) => {
-                sibling.classList.add('hidden');
-            });
-        } else {
-            arrow.classList.add('rotate-90');
-            current.dataset.ariaExpanded = 'true';
-
-            nextSiblings.forEach((sibling) => {
-                sibling.classList.remove('hidden');
-            });
-        }
-    }
-
-    /*!
-     * Get all following siblings of each element up to but not including the element matched by the selector
-     * (c) 2017 Chris Ferdinandi, MIT License, https://gomakethings.com
-     * @param  {Node}   elem     The element
-     * @param  {String} selector The selector to stop at
-     * @param  {String} filter   The selector to match siblings against [optional]
-     * @return {Array}           The siblings
+    /**
+     * @param {Event} event
      */
-    nextUntil(elem, selector, filter) {
-
-        // Setup siblings array
-        var siblings = [];
+    toggle(event)
+    {
+        /** @type {HTMLElement[]} */
+        const contentTargets = this.contentTargets;
 
-        // Get the next sibling element
-        elem = elem.nextElementSibling;
+        /** @type {HTMLElement[]} */
+        const headerTargets = this.headerTargets;
 
-        // As long as a sibling exists
-        while (elem) {
+        /** @type {HTMLElement[]} */
+        const arrowTargets = this.arrowTargets;
 
-            // If we've reached our match, bail
-            if (elem.matches(selector)) break;
+        /** @type {string} */
+        const arrowRotateClassString = this.arrowRotateClasses;
 
-            // If filtering by a selector, check if the sibling matches
-            if (filter && !elem.matches(filter)) {
-                elem = elem.nextElementSibling;
-                continue;
-            }
-
-            // Otherwise, push it to the siblings array
-            siblings.push(elem);
-
-            // Get the next sibling element
-            elem = elem.nextElementSibling;
+        /** @type {string} */
+        const contentHiddenClassString = this.contentHiddenClasses;
 
+        const current = event.currentTarget;
+        const header = this.closestTarget(current, 'header');
+        const arrow = this.closestTarget(current, 'arrow');
+        const content = header.nextElementSibling;
+
+        const isOpen = header.getAttribute('aria-expanded') === 'true';
+
+        // Reset all accordions
+        headerTargets.forEach((header) => {
+            header.setAttribute('aria-expanded', 'false');
+        });
+        arrowTargets.forEach((arrow) => {
+            arrow.classList.remove(arrowRotateClassString);
+        });
+        contentTargets.forEach((content) => {
+            content.classList.add(contentHiddenClassString);
+        });
+
+        // Open the current accordion if it wasn't open before (else we toggle it)
+        if (!isOpen) {
+            arrow.classList.add(arrowRotateClassString);
```

Comment:
If using the `<x>Classes` instead of `<x>Class`, it's an array. Therefore, this would be a spread:

```suggestion
            arrow.classList.add(...arrowRotateClassString);
```